### PR TITLE
[CLI] Add qmk-hid bootloader detection support to `qmk console`

### PR DESCRIPTION
**Pull request by drashna**
## Description

Like the title says.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/console.py [hunk 1]
```diff
@@ -51,7 +51,8 @@
     ('239A', '000E'): 'caterina: Adafruit ItsyBitsy 32U4 5v',
```
Comment (fauxpark) [inline lib/python/qmk/cli/console.py:51]:
```suggestion
```
Let's remove this duplicate entry too.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: util/udev/50-qmk.rules [hunk 2]
```diff
@@ -63,3 +63,9 @@ SUBSYSTEMS=="usb", ATTRS{idVendor}=="2a03", ATTRS{idProduct}=="0037", TAG+="uacc
 
 # hid_listen
 KERNEL=="hidraw*", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl"
+
+# hid bootloaders
+## QMK HID
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2076", TAG+="uaccess"
+## PJRC's HalfKay
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="0483", TAG+="uaccess"
```
Comment (sigprof) [inline util/udev/50-qmk.rules:71]:
```suggestion
SUBSYSTEMS=="usb", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="0478", TAG+="uaccess"
```

**Review by sigprof — CHANGES_REQUESTED** [review]
The USB ID for Halfkay in the udev rules is wrong — that bootloader actually uses `16C0:0478` (and the ID is correct in other places).

File: util/udev/50-qmk.rules [hunk 3]
```diff
@@ -63,3 +63,9 @@ SUBSYSTEMS=="usb", ATTRS{idVendor}=="2a03", ATTRS{idProduct}=="0037", TAG+="uacc
 
 # hid_listen
 KERNEL=="hidraw*", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl"
+
+# hid bootloaders
+## QMK HID
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2076", TAG+="uaccess"
```
Comment (sigprof) [inline util/udev/50-qmk.rules:69]:
```suggestion
SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2067", TAG+="uaccess"
```

Comment (sigprof) [reply util/udev/50-qmk.rules:69]: Verified the file with this fix — works for flashing an ATmega32U4 with nanoBoot (which uses the same USB IDs and flashing tools as the LUFA/QMK HID bootloader).

Comment (skullydazed) [reply util/udev/50-qmk.rules:69]: We've just started the process of migrating our udev rules to a dedicated repo and tool, could we make sure the same changes are applied there? 

https://github.com/qmk/qmk_udev

Comment (drashna) [reply util/udev/50-qmk.rules:69]: @skullydazed that was the plan. :)
But only so much I can do at 1-2AM. :D

**Review by sigprof — APPROVED** [review]
Verified the udev rules on Linux (NixOS x86_64) using `qmk flash` with Teensy++ 2.0 (halfkay), nanoBoot (using the same 03eb:2067 ID as QMK/LUFA HID), and also stm32-dfu on STM32F411 and atmel-dfu on atmega32u4; everything works as it should.

**Review by tzarc — APPROVED** [review]

**Pull request merged**